src/FSharp.Analyzers.Cli/Logging.fs

```diff
@@ -0,0 +1,79 @@
+module FSharp.Analyzers.Cli.CustomLogging
+
+open System
+open System.IO
+open System.Runtime.CompilerServices
+open Microsoft.Extensions.Logging
+open Microsoft.Extensions.Logging.Console
+open Microsoft.Extensions.Logging.Abstractions
+open Microsoft.Extensions.Options
+
+type CustomOptions() =
+    inherit ConsoleFormatterOptions()
+
+    /// if true: no LogLevel as prefix, colored output according to LogLevel
+    /// if false: LogLevel as prefix, no colored output
+    member val UseAnalyzersMsgStyle = false with get, set
+
+type CustomFormatter(options: IOptionsMonitor<CustomOptions>) as this =
+    inherit ConsoleFormatter("customName")
+
+    let mutable optionsReloadToken: IDisposable = null
+    let mutable formatterOptions = options.CurrentValue
+    let origColor = Console.ForegroundColor
+
+    do optionsReloadToken <- options.OnChange(fun x -> this.ReloadLoggerOptions(x))
+
+    member private _.ReloadLoggerOptions(opts: CustomOptions) = formatterOptions <- opts
+
+    override this.Write<'TState>
+        (
+            logEntry: inref<LogEntry<'TState>>,
+            _scopeProvider: IExternalScopeProvider,
+            textWriter: TextWriter
+        )
+        =
+        let message = logEntry.Formatter.Invoke(logEntry.State, logEntry.Exception)
+
+        if formatterOptions.UseAnalyzersMsgStyle then
+            this.SetColor(textWriter, logEntry.LogLevel)
+            textWriter.WriteLine(message)
+            this.ResetColor()
+        else
+            this.WritePrefix(textWriter, logEntry.LogLevel)
+            textWriter.WriteLine(message)
+
+    member private _.WritePrefix(textWriter: TextWriter, logLevel: LogLevel) =
+        match logLevel with
+        | LogLevel.Trace -> textWriter.Write("trace: ")
+        | LogLevel.Debug -> textWriter.Write("debug: ")
+        | LogLevel.Information -> textWriter.Write("info: ")
+        | LogLevel.Warning -> textWriter.Write("warn: ")
+        | LogLevel.Error -> textWriter.Write("error: ")
+        | LogLevel.Critical -> textWriter.Write("critical: ")
+        | _ -> ()
+
+    member private _.SetColor(textWriter: TextWriter, logLevel: LogLevel) =
+        let color =
+            match logLevel with
```

Comment:
This seems so weird that this is the way you need to do this.
Can you put in any documentation reference?

Reply:
I think the lack of a color setting API in the `TextWriter` makes this the way to go.